pkg/cstorvolume/v1alpha1/build.go

```diff
@@ -0,0 +1,275 @@
+	}
+	return b
+}
+
+// WithLabelsNew resets existing labels if any with
+// ones that are provided here
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: no new labels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.cstorvolume.object.Labels = newlbls
+	return b
+}
+
+// WithTargetIP sets the target IP address field of
+// CStorVolume with provided arguments
+func (b *Builder) WithTargetIP(targetip string) *Builder {
+	if len(targetip) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing targetip"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.TargetIP = targetip
+	return b
+}
+
+// WithCapacity sets the Capacity field of CStorVolume with provided arguments
+func (b *Builder) WithCapacity(capacity string) *Builder {
+	if len(capacity) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing capacity"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.Capacity = capacity
+	return b
+}
+
+// WithNodeBase sets the NodeBase field of CStorVolume with provided arguments
+func (b *Builder) WithNodeBase(nodebase string) *Builder {
+	if len(nodebase) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing nodebase"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.NodeBase = nodebase
+	return b
+}
+
+// WithIqn sets the Iqn field of CStorVolume with provided arguments
+func (b *Builder) WithIqn(iqn string) *Builder {
+	if len(iqn) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing iqn"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.Iqn = iqn
+	return b
+}
+
+// WithTargetPort sets the TargetPort field of
+// CStorVolume with provided arguments
+func (b *Builder) WithTargetPort(targetport string) *Builder {
+	if len(targetport) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing targetport"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.TargetPort = targetport
+	return b
+}
+
+// WithTargetPortal sets the TargetPortal field of
+// CStorVolume with provided arguments
+func (b *Builder) WithTargetPortal(targetportal string) *Builder {
+	if len(targetportal) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing targetportal"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.TargetPortal = targetportal
+	return b
+}
+
+// WithReplicationFactor sets the ReplicationFactor field of
+// CStorVolume with provided arguments
+func (b *Builder) WithReplicationFactor(replicationfactor int) *Builder {
+	if replicationfactor <= 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: invalid replicationfactor"),
```

Comment:
let us also provide the value of replicationfactor in the message.
Lets try to think from end user perspective.

Reply:
Done